lib/bamboozled/api/report.rb

```diff
@@ -6,30 +6,36 @@ def find(number, format = "JSON", fd_param = true)
         request(:get, "reports/#{number}?format=#{format.upcase}&fd=#{fd_param.yesno}")
       end
 
-      def custom(fields, include_null = true ,date = nil,  format='JSON' )
-        url = URI("#{path_prefix}reports/custom?format=#{format.upcase}")
-        puts url
-        http = Net::HTTP.new(url.host, url.port)
-        http.use_ssl = true
-        request = Net::HTTP::Post.new(url)
-        request["content-type"] = 'application/json'
-        request.basic_auth auth[:username], auth[:password]
-        request.body = body(fields, include_null, date).to_json
-        response = http.request(request)
-        response.read_body
+      def custom(fields, include_null = true, last_changed_date = nil,  format = "JSON")
+        options = {
+          body: body(fields, include_null, last_changed_date),
+          headers: { "Content-Type" => "application/json" }
+        }
+
+        response = request(:post, "reports/custom?format=#{format.upcase}", options)
+        response["employees"]
       end
 
       private
 
-      def body(fields, include_null, date)
+      def body(fields, include_null, last_changed_date)
         body = {}
-        body[:fields] = fields || FieldCollection.all_names
-        last_changed_object = {}
-        last_changed_object[:includeNull] = include_null.to_s
-        last_changed_object[:value] = date
+        body[:fields] = if fields == :all
```

Comment:
Maybe the if clause could be in a separate method. Just a suggestion. No blocker 😊

Reply:
Thanks for the suggestion! I'll pass this time and prioritize functionality given the context of this project